app/src/main/java/swati4star/createpdf/fragment/FavouritesFragment.java

```diff
@@ -0,0 +1,203 @@
+import android.support.v7.preference.PreferenceManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import swati4star.createpdf.R;
+import swati4star.createpdf.activity.FavouritesActivity;
+import swati4star.createpdf.customviews.MyCardView;
+
+public class FavouritesFragment extends Fragment
+        implements SharedPreferences.OnSharedPreferenceChangeListener {
+
+    private Activity mActivity;
+    private SharedPreferences mSharedPreferences;
+
+    @BindView(R.id.fav_add_fab)
+    FloatingActionButton mFab;
+    @BindView(R.id.images_to_pdf_fav)
+    MyCardView pref_img_to_pdf;
+    @BindView(R.id.text_to_pdf_fav)
+    MyCardView pref_text_to_pdf;
+    @BindView(R.id.qr_barcode_to_pdf_fav)
+    MyCardView pref_qr_barcode;
+    @BindView(R.id.view_files_fav)
+    MyCardView pref_view_files;
+    @BindView(R.id.view_history_fav)
+    MyCardView pref_history;
+    @BindView(R.id.add_password_fav)
+    MyCardView pref_add_password;
+    @BindView(R.id.remove_password_fav)
+    MyCardView pref_rem_pass;
+    @BindView(R.id.rotate_pages_fav)
+    MyCardView pref_rot_pages;
+    @BindView(R.id.add_watermark_fav)
+    MyCardView pref_add_watermark;
+    @BindView(R.id.add_images_fav)
+    MyCardView pref_add_images;
+    @BindView(R.id.merge_pdf_fav)
+    MyCardView pref_merge_pdf;
+    @BindView(R.id.split_pdf_fav)
+    MyCardView pref_split_pdf;
+    @BindView(R.id.invert_pdf_fav)
+    MyCardView pref_invert_pdf;
+    @BindView(R.id.compress_pdf_fav)
+    MyCardView pref_compress;
+    @BindView(R.id.remove_duplicates_pages_pdf_fav)
+    MyCardView pref_rem_dup_pages;
+    @BindView(R.id.remove_pages_fav)
+    MyCardView pref_remove_pages;
+    @BindView(R.id.rearrange_pages_fav)
+    MyCardView pref_reorder_pages;
+    @BindView(R.id.extract_images_fav)
+    MyCardView pref_extract_img;
+    @BindView(R.id.pdf_to_images_fav)
+    MyCardView pref_pdf_to_img;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootview = inflater.inflate(R.layout.favourites_fragment, container, false);
+        ButterKnife.bind(this, rootview);
+        SharedPreferences sharedPreferences = PreferenceManager
+                .getDefaultSharedPreferences(getActivity());
+        sharedPreferences.registerOnSharedPreferenceChangeListener(this);
+        ActionBar actionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
+        actionBar.hide();
+        mFab.setOnClickListener(v ->
+                startActivity(new Intent(this.getContext(), FavouritesActivity.class))
+        );
+        return rootview;
+    }
+
+    private void checkFavs(SharedPreferences sharedPreferences) {
+        if (sharedPreferences.getBoolean(getResources().getString(R.string.img_to_pdf),
```

Comment:
too much code repitition here. Why dont you create a function that takes the required parameters and do this task. Will make the code easy to understand also

Reply:
@Swati4star I thought about it but then I need to create Hash maps and run throughout it in a loop as I need to check for every checked/unchecked favourite

Reply:
@Swati4star @srv-twry Any thoughts on my above comment?

Reply:
You can move this part in a function, which takes stringid and view as parameteres 
```
if (sharedPreferences.getBoolean(getResources().getString(stringid),
                false)) {
            view.setVisibility(View.VISIBLE);
        } else {
            view.setVisibility(View.GONE);
        }
```
Now, if you wish you can call function for every view or create a simple array of views & string ids, and just traverse those

Reply:
@Swati4star Can you please tell me your thoughts about the above idea?

Reply:
I have already replied and given you an alternate to hashmap

Reply:
Okay @Swati4star . So then that function needs to be called multiple times right?

Reply:
yes. 

> call function for every view or create a simple array of views & string ids, and just traverse those